src/ucp/dt/datatype_iter.inl

```diff
@@ -296,7 +296,7 @@ ucp_datatype_iter_mem_reg(ucp_context_h context, ucp_datatype_iter_t *dt_iter,
     ucs_assert(dt_iter->dt_class == UCP_DATATYPE_CONTIG);
     return ucp_mem_rereg_mds(context, md_map, dt_iter->type.contig.buffer,
                              dt_iter->length, UCT_MD_MEM_ACCESS_RMA, NULL,
-                             dt_iter->mem_type, NULL,
+                             (ucs_memory_type_t)dt_iter->mem_info.type, NULL,
```

Comment:
Based on this https://github.com/openucx/ucx/pull/6178/ #discussion_r563083876 do we want to pass mem_info into ucp_mem_rereg_mds instead of just mem_type?

Reply:
same as https://github.com/openucx/ucx/pull/6180#discussion_r567247764